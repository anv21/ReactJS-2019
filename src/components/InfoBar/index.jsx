import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, Switch, Link } from 'react-router-dom';

import SORT_BY from '../../constants/SORT_BY';
import { setSortBy } from '../../actions';

class InfoBar extends React.Component {
    onSortByReleaseDateClick = () => {
        this.props.setSortBy(SORT_BY.RELEASE_DATE);
    }

    onSortByRatingClick = () => {
        this.props.setSortBy(SORT_BY.RATING);
    }

    render() {
        const {value, items, item, total, sortBy, searchBy} = this.props;

        return (
            <Switch>
                <Route path='/film' render={() => (
                    <div className="infoBar_section">
                        {items && items.length > 0 &&
                            <div className="infoBar_wrapper">
                                <label className="infoBar_label">
                                    Films by {item && item.genres && item.genres[0]}
                                </label>
                            </div>
                        }
                    </div>
                )}
                />
                <Route path='/' render={() => (
                    <div className="infoBar_section">
                        {items && items.length > 0 &&
                            <div className="infoBar_wrapper">
                                <label className="infoBar_label">
                                    {total} movies found
                                </label>
                                <div>
                                    <label className="infoBar_label">Sort by</label>
                                    <Link to={`/search?search=${value}&searchBy=${searchBy}&sortBy=${SORT_BY.RELEASE_DATE}`}>
                                        <span className={"infoBar_sort_by_link " + (sortBy === SORT_BY.RELEASE_DATE && "active_link")}
                                            onClick={this.onSortByReleaseDateClick}>
                                            release date
                                        </span>
                                    </Link>
                                    <Link to={`/search?search=${value}&searchBy=${searchBy}&sortBy=${SORT_BY.RATING}`}>
                                        <span className={"infoBar_sort_by_link " + (sortBy === SORT_BY.RATING && "active_link")}
                                            onClick={this.onSortByRatingClick}>
                                            rating
                                        </span>
                                    </Link>
                                </div>
                            </div>
                        }
                    </div>)}
                />
            </Switch>
        )
    }
}

InfoBar.propTypes = {
    value: PropTypes.string,
    items: PropTypes.array,
    item: PropTypes.object,
    total: PropTypes.number,
    sortBy: PropTypes.string,
    searchBy: PropTypes.string,
    setSortBy: PropTypes.func,
};

InfoBar.defaultProps = {
    value: '',
    items: [],
    item: {},
    total: 0,
    sortBy: '',
    searchBy: '',
    setSortBy: null,
};

const mapStateToProps = (state) => {
    const {value, items, item, total, sortBy, searchBy} = state.appReducer;
    return {value, items, item, total, sortBy, searchBy};
};

export default connect(mapStateToProps, {setSortBy})(InfoBar);