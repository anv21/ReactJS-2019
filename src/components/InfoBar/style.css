.infoBar_section {
    display: block;
    width: 100%;
    height: 50px;
    background-color: #f5f5f5;
}

.infoBar_wrapper {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 50px;
    width: 900px;
    margin: 0 auto;
}

.infoBar_label {
    color: #0e0e0e;
}

.infoBar_sort_by_link {
    margin: 5px 5px 5px 20px;
    color: #0e0e0e;
    text-transform: none;
}

.infoBar_sort_by_link:hover {
    color:#f55263;
    cursor: pointer;
}

.active_link {
    color:#f55263;
}
