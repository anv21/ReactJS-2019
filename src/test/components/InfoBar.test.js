import React from 'react';
import { shallow } from 'enzyme';
import {mockData} from '../../mock-data/data.json'

import InfoBar from '../../components/InfoBar/index.jsx';

import APP_STATES from '../../constants/APP_STATES';

const items = mockData;

describe('InfoBar', () => {
    const infoBar = shallow(<InfoBar/>);

    it('should be like snapshot when AppState is SearchPage', () => {
        infoBar.setProps({appState: APP_STATES.SEARCH_PAGE, items});
        expect(infoBar).toMatchSnapshot();
    });

    it('should be like snapshot when AppState is DetailsPage', () => {
        infoBar.setProps({appState: APP_STATES.DETAILS_PAGE, items});
        expect(infoBar).toMatchSnapshot();
    });
});