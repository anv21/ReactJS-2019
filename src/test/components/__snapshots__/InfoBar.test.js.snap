// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar should be like snapshot when AppState is DetailsPage 1`] = `ShallowWrapper {}`;

exports[`InfoBar should be like snapshot when AppState is SearchPage 1`] = `ShallowWrapper {}`;
